cli/coverage.rs

```diff
@@ -277,48 +265,57 @@ impl PrettyCoverageReporter {
           if range.start_offset <= line_start_offset
             && range.end_offset >= line_end_offset
           {
-            count += range.count;
             if range.count == 0 {
               count = 0;
               break;
             }
+
+            count += range.count;
           }
         }
-      }
 
+        line_start_offset = line_end_offset;
+      }
       if count > 0 {
-        covered_lines += 1;
+        covered_lines.push(index);
+      } else {
+        uncovered_lines.push(index);
       }
-
-      total_lines += 1;
-      line_offset += line.len();
     }
 
-    let line_ratio = covered_lines as f32 / total_lines as f32;
-    let line_coverage = format!("{:.3}%", line_ratio * 100.0);
+    if !self.quiet {
+      print!("cover {} ... ", coverage.script_coverage.url);
 
-    let line = if line_ratio >= 0.9 {
-      format!(
-        "{} {}",
-        coverage.script_coverage.url,
-        colors::green(&line_coverage)
-      )
-    } else if line_ratio >= 0.75 {
-      format!(
-        "{} {}",
-        coverage.script_coverage.url,
-        colors::yellow(&line_coverage)
-      )
-    } else {
-      format!(
-        "{} {}",
-        coverage.script_coverage.url,
-        colors::red(&line_coverage)
-      )
-    };
+      let line_coverage_ratio = covered_lines.len() as f32 / lines.len() as f32;
+      let line_coverage = format!(
+        "{:.3}% ({}/{})",
+        line_coverage_ratio * 100.0,
+        covered_lines.len(),
+        lines.len()
+      );
 
-    Some(line)
+      if line_coverage_ratio >= 0.9 {
+        println!("{}", colors::green(&line_coverage));
+      } else if line_coverage_ratio >= 0.75 {
+        println!("{}", colors::yellow(&line_coverage));
+      } else {
+        println!("{}", colors::red(&line_coverage));
+      }
+
+      for line_index in uncovered_lines {
+        let line = format!(
+            "{:width$} {}",
+            line_index + 1,
+            lines[line_index],
+            width = 4
+        );
+
+        println!("{}", colors::red(&line));
+      }
+    }
   }
+
+  pub fn close(&self) {}
```

Comment:
Final summary of all coverage will (eventually) go here; just haven't figured out what that looks like yet.